firmware/boron-ins-fsm/test/mocks/System.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+class MockSystem
+{
+public:
+    MockSystem()
+    {
+    }
+
+public:
+    bool reset() const
+    {
+        printf("System Reset Works");
+        return true;
```

Comment:
It would be really nice if there was some way that we could tell that this function was being called or not. 

Is it possible to add an extern variable in this file for `resetWasCalled`? 
Then
1. Before each test, you can set it to false
2. In this mocked function you will set it to true
3. Then in the test you can check that it is now true, thus showing that the function was, in fact, called

I don't think we're doing this anywhere else yet, but it feels like it should work. Could you please try it out? Let me know if my description isn't clear enough!

Reply:
If this works, it would be amazing if you could extend the `mocks/Particle.h` file's `publish` function to do something similar, but instead to put the whole string into the `extern` variable so it can be checked no only that it was called but that it was called with the expected parameters.

Reply:
It worked for the reset, so ill do it for the publish as well